quactography/solver/qaoa_solver_qu_edge.py

```diff
@@ -1,235 +1,238 @@
-
-    # Save parameters alpha and min_cost with path in csv file:
-    opt_path = str_path_reversed
-
-    save_optimization_results(
-        dist=dist,
-        dist_binary_probabilities=dist_binary_probabilities,
-        min_cost=min_cost,
-        hamiltonian=h,
-        outfile=outfile,
-        # It is reversed as classical read to be compared to exact_path code when diagonalising Hamiltonian
-        opt_bin_str=opt_path,
-        reps=reps,
-        opt_params=resx  # type: ignore
-    )  # type: ignore
-
-
-def multiprocess_qaoa_solver_edge(
-    hamiltonians,
-    reps,
-    nbr_processes,
-    output_file,
-    optimizer,
-    cost_landscape,
-    save_only,
-):
-    """
-    Solve the optimization problem using the QAOA algorithm
-    with multiprocessing on the alpha values.
-
-    Parameters
-    ----------
-    hamiltonians : list
-        List of Hamiltonian objects from quactography library, Hamiltonian_qubit_edge.
-    batch_count : int
-        Number of time the command will be ran
-    reps : int
-        Number of repetitions for the QAOA algorithm,
-        determines the number of sets of gamma and beta angles.
-    nbr_processes : int
-        Number of cpu to use for multiprocessing. default=1
-    output_file : str
-        The output file name for the optimization results in .npz format.
-    optimizer : str
-        Optimizer to use for the QAOA algorithm. default="Differential"
-    cost_landscape : bool
-        Plot the cost landscape with the optimal point if reps=1. default=False
-    save_only : bool
-        If True, the figure is saved without displaying it. default=False
-
-    Returns
-    -------
-    None
-    """
-    pool = multiprocessing.Pool(nbr_processes)
-
-    pool.map(
-        find_longest_path,
-        zip(
-            hamiltonians,
-            itertools.repeat(reps),
-            itertools.repeat(output_file),
-            itertools.repeat(optimizer),
-            itertools.repeat(cost_landscape),
-            itertools.repeat(save_only),
-        ),
-    )
-    pool.close()
-    pool.join()
-
-    print(
-        "------------------MULTIPROCESS SOLVER FINISHED-------------------------"
-    )
+import multiprocessing
+import itertools
+
+from qiskit.primitives import Estimator, Sampler
+from qiskit.circuit.library import QAOAAnsatz
+import matplotlib.pyplot as plt
+import numpy as np
+from qiskit.quantum_info import SparsePauliOp
+from scipy.optimize import differential_evolution
+from functools import partial
+
+
+from quactography.solver.io import save_optimization_results
+from quactography.solver.optimization_loops import (
+    POWELL_loop_optimizer,
+    POWELL_refinement_optimization,
+)
+from quactography.visu.plot_cost_landscape import plt_cost_func
+
+# !!!!!!!!!!! Optimal path returned from this optimisation is in
+# classical read (meaning the zeroth qubit is the first from left
+# term in binary string) !!!!!!!!!!!!!!!
+
+alpha_min_costs = []
+
+
+# Minimization cost function

```

Comment:
You can remove the comments explaining the functions since you have well documented the docstring.